src/compiler/utils/get_code_frame.rs

```diff
@@ -0,0 +1,28 @@
+use regex::{Regex, Captures};
+use std::cmp::{max, min};
+
+fn tabs_to_spaces(str: &str) -> String {
+    let re = Regex::new(r"^\t+").unwrap();
+    re.replace_all(str, |caps: &Captures| {
+        let matched = caps.get(0).unwrap();
```

Comment:
unsafe code here.

Reply:
Wait I am confused how is that unsafe?

Reply:
How can we make sure that we always capture the value? I suggest using and_then or map instead. We should use as least unwrap as possible.

Reply:
@linrium According to the [docs]:

> The 0th capture always corresponds to the entire match

So it's appropriate to use unwrap here. Heck, this could be shorten down to:

```rust
let matched = &caps[0];
```

On a side note, see [this recent blog post][unwrap] on `unwrap` usage in Rust.

[docs]: https://docs.rs/regex/latest/regex/struct.Captures.html
[unwrap]: https://blog.burntsushi.net/unwrap/

Reply:
Hopefully the changes I made resolves it. Changed the unwrap method to just a reference to the 0th capture.

Reply:
@iceghost ice thanks TIL.